include/rspq.h

```diff
@@ -0,0 +1,713 @@
+ * going to write a number of subsequent commands in straight line code, you
+ * can postpone the call to #rspq_flush after the whole sequence has been written.
+ * 
+ * @code{.c}
+ * 		// This example shows some code configuring the lights for a scene.
+ * 		// The command in this sample is called CMD_SET_LIGHT and requires
+ * 		// a light index and the RGB colors for the list to update.
+ *      uint32_t gfx_overlay_id;   
+ *
+ * 		#define CMD_SET_LIGHT  0x7
+ *
+ * 		for (int i=0; i<MAX_LIGHTS; i++) {
+ * 			rspq_write(gfx_overlay_id, CMD_SET_LIGHT, i,
+ * 			    (lights[i].r << 16) | (lights[i].g << 8) | lights[i].b);
+ * 		}
+ * 		
+ * 		// After enqueuing multiple commands, it is sufficient
+ * 		// to call rspq_flush once to make sure the RSP runs them (in case
+ * 		// it was idling).
+ * 		rspq_flush();
+ * @endcode
+ *
+ * @note This is an experimental API. In the future, it might become 
+ *       a no-op, and flushing could happen automatically at every #rspq_write.
+ *       We are keeping it separate from #rspq_write while experimenting more
+ *       with the RSPQ API.
+ * 
+ * @note This function is a no-op if it is called while a block is being recorded
+ *       (see #rspq_block_begin / #rspq_block_end). This means calling this function
+ *       in a block recording context will not guarantee the execution of commands
+ *       that were queued prior to starting the block.
+ *       
+ */
+void rspq_flush(void);
+
+/**
+ * @brief Wait until all commands in the queue have been executed by RSP.
+ *
+ * This function blocks until all commands present in the queue have
+ * been executed by the RSP and the RSP is idle.
+ * 
+ * This function exists mostly for debugging purposes. Calling this function
+ * is not necessary, as the CPU can continue adding commands to the queue
+ * while the RSP is running them. If you need to synchronize between RSP and CPU
+ * (eg: to access data that was processed by RSP) prefer using #rspq_syncpoint /
+ * #rspq_wait_syncpoint which allows for more granular synchronization.
+ */
+#define rspq_wait() ({ \
+    rspq_wait_syncpoint(rspq_syncpoint()); \
+})
+
+/**
+ * @brief Create a syncpoint in the queue.
+ * 
+ * This function creates a new "syncpoint" referencing the current position
+ * in the queue. It is possible to later check when the syncpoint
+ * is reached by the RSP via #rspq_check_syncpoint and #rspq_wait_syncpoint.
+ *
+ * @return     ID of the just-created syncpoint.
+ * 
+ * @note It is not possible to create a syncpoint within a block because it
+ *       is meant to be a one-time event. Otherwise the same syncpoint would
+ *       potentially be triggered multiple times, which is not supported.
+ * 
+ * @note It is not possible to create a syncpoint from the high-priority queue
+ *       due to the implementation requiring syncpoints to be triggered
+ *       in the same order they have been created.
+ * 
+ * @see #rspq_syncpoint_t
+ */
+rspq_syncpoint_t rspq_syncpoint(void);
```

Comment:
I did a final pass on rspq.h and I suggest that we rename this function. I find it a bit confusing to have both `rspq_syncpoint` and `rspq_syncpoint_t`. I would say that this should be called `rspq_syncpoint_new()`. Also, we should probably rename the other functions into `rspq_syncpoint_check` and `rspq_syncpoint_wait`, to adhere to the convention of having a hierarchical prefix.

Reply:
I renamed those functions like you suggested